File: 1209. Remove All Adjacent Duplicates in String II.cpp
class Solution {
public:
    string removeDuplicates(string s, int k) {
        vector<pair<char,int>> stk;
        for (auto c : s) {
            if (!stk.empty() && stk.back().first==c) 
                ++stk.back().second;
            else 
                stk.push_back({c, 1});
            if (stk.back().second==k)
                stk.pop_back();
        }
        string ans;
        for (auto& pr : stk)
            ans += string(pr.second, pr.first);
        return ans;
    }
};

File: 1209. Remove All Adjacent Duplicates in String II.py
class Solution:
    def removeDuplicates(self, s: str, k: int) -> str:
        stk = []
        for c in s:
            if stk and stk[-1][0]==c:
                stk[-1][1] += 1
            else:
                stk.append([c, 1])
            if stk[-1][1]==k:
                stk.pop()
        return ''.join([c*x for c,x in stk])